tests/rptest/tests/random_node_operations_test.py

```diff
@@ -217,23 +242,31 @@ def verify(self):
             del self.consumer
 
             # Start a new consumer to read all data written
-            self._start_consumer()
+            self._start_consumer(with_logs=True)
             self.consumer.wait()
-            if self.topic.cleanup_policy != TopicSpec.CLEANUP_COMPACT:
+            if not self.compaction_enabled:
                 assert self.consumer.consumer_status.validator.valid_reads >= self.producer.produce_status.acked, f"Missing messages from topic: {self.topic}. valid reads: {self.consumer.consumer_status.validator.valid_reads}, acked messages: {self.producer.produce_status.acked}"
 
             assert self.consumer.consumer_status.validator.invalid_reads == 0, f"Invalid reads in topic: {self.topic}, invalid reads count: {self.consumer.consumer_status.validator.invalid_reads}"
 
     @skip_debug_mode
     @cluster(num_nodes=7,
              log_allow_list=CHAOS_LOG_ALLOW_LIST + PREV_VERSION_LOG_ALLOW_LIST)
-    @matrix(enable_failures=[False, True],
+    @matrix(enable_failures=[True, False],
             num_to_upgrade=[0],
-            enable_controller_snapshots=[True, False])
+            enable_controller_snapshots=[True, False],
+            with_tiered_storage=[True, False])
     def test_node_operations(self, enable_failures, num_to_upgrade,
-                             enable_controller_snapshots):
+                             enable_controller_snapshots, with_tiered_storage):
 
         lock = threading.Lock()
+        default_segment_size = 1024 * 1024
+
+        # do not test controller snapshots with older versions
+        # as they do not support it
```

Comment:
nit: I think if we were to upgrade, this would begin with the previous version as 23.2 -- would controller snapshots work then without additional changes?